### ISSUE TITLE
Docker not producing any logs

### ISSUE THREAD

**Issue by vishr** [mheese/journalbeat#93]
Hi,

I am running journalbeat docker on ubuntu 17.04 but I don't see any logs in the console or logstash, however, `journalctl -f` do produce them on the host. Any ideas?

```sh
$ docker run -d -v /var/log/journal:/var/log/journal \
-v /run/log/journal:/run/log/journal \
-v /etc/machine-id:/etc/machine-id \
my-journalbeat
```

Dockerfile

```Dockerfile
FROM mheese/journalbeat

COPY journalbeat.yml .

CMD ["./journalbeat", "-e", "-c", "journalbeat.yml"]
```

Config

```yaml
journalbeat:
  seek_position: cursor
  cursor_seek_fallback: tail
  write_cursor_state: true
  cursor_state_file: /data/cursor
  pending_queue.file: /data/pending
  move_metadata_to_field: journal

# output.logstash:
#   enabled: true
#   hosts: ["logstash:5044"]

output.console:
  enabled: true
  pretty: false

logging.to_files: false
```

**Comment by hookenz** [comment]
Run it in debug mode like this and look for errors

     $ docker run -d -v /var/log/journal:/var/log/journal \
        -v /run/log/journal:/run/log/journal \
        -v /etc/machine-id:/etc/machine-id \
        my-journalbeat -e -d "*"

I suspect the path /data does not exist in the container so cursor_state_file and pending_queue.file are not able to be written.  Try moving them to / or add `RUN mkdir /data` to your `Dockerfile`

**Comment by vishr** [comment]
@hookenz thanks for your response.

`/data` exists in the docker. I ran with the debug on and got this response:

```sh
2017/08/17 00:56:03.190944 beat.go:285: INFO Home path: [/] Config path: [/] Data path: [//data] Logs path: [//logs]
2017/08/17 00:56:03.190960 beat.go:186: INFO Setup Beat: journalbeat; Version: 5.5.0
2017/08/17 00:56:03.190970 processor.go:44: DBG  Processors: 
2017/08/17 00:56:03.190980 beat.go:192: DBG  Initializing output plugins
2017/08/17 00:56:03.191006 outputs.go:108: INFO Activated console as output plugin.
2017/08/17 00:56:03.191015 publish.go:238: DBG  Create output worker
2017/08/17 00:56:03.191028 metrics.go:23: INFO Metrics logging every 30s
2017/08/17 00:56:03.191058 publish.go:280: DBG  No output is defined to store the topology. The server fields might not be filled.
2017/08/17 00:56:03.191087 publish.go:295: INFO Publisher name: journalbeat1
2017/08/17 00:56:03.191200 async.go:63: INFO Flush Interval set to: 1s
2017/08/17 00:56:03.191207 async.go:64: INFO Max Bulk Size set to: 2048
2017/08/17 00:56:03.191215 async.go:72: DBG  create bulk processing worker (interval=1s, bulk size=2048)
2017/08/17 00:56:03.192080 journalbeat.go:97: WARN Could not seek to cursor: reading cursor state file failed: open /data/cursor: no such file or directory
2017/08/17 00:56:03.192096 journalbeat.go:52: INFO Seek to tail successful
2017/08/17 00:56:03.192169 beat.go:221: INFO journalbeat start running.
2017/08/17 00:56:03.192181 journalbeat.go:172: INFO Journalbeat is running!
2017/08/17 00:56:03.192214 journalbeat.go:191: WARN could not read the pending queue: open /data/pending: no such file or directory
2017/08/17 00:56:33.191618 metrics.go:34: INFO No non-zero metrics in the last 30s
2017/08/17 00:57:03.191313 metrics.go:34: INFO No non-zero metrics in the last 30s
2017/08/17 00:57:33.191610 metrics.go:34: INFO No non-zero metrics in the last 30s
2017/08/17 00:58:03.191595 metrics.go:34: INFO No non-zero metrics in the last 30s
```

**Comment by mbrooks** [comment]
I don't know why it's not logging to console, but it's not making to logstash because you have no logstash servers defined. We use the following config for the default journal dockerfile, maybe it will help?

https://github.com/mheese/journalbeat/blob/master/docker/journalbeat.yml#L13-L15

**Comment by vishr** [comment]
@mbrooks thanks for your response. I have logstash server, it is just commented - I have tried logstash, elasticsearch and console but I am not able to get data anywhere.

**Comment by c4milo** [comment]
I'm also trying to run journalbeat within a container, I mounted all the necessary volumes but I can't tell if it is working.  

```
9/01 17:53:51.082402 beat.go:285: INFO Home path: [/] Config path: [/etc/journalbeat] Data path: [/var/lib/journalb
9/01 17:53:51.082454 beat.go:186: INFO Setup Beat: journalbeat; Version: 5.5.1
9/01 17:53:51.082470 processor.go:44: DBG  Processors:
9/01 17:53:51.082483 beat.go:192: DBG  Initializing output plugins
9/01 17:53:51.082514 outputs.go:108: INFO Activated console as output plugin.
9/01 17:53:51.082529 publish.go:238: DBG  Create output worker
9/01 17:53:51.082665 publish.go:280: DBG  No output is defined to store the topology. The server fields might not b
9/01 17:53:51.082799 publish.go:295: INFO Publisher name: journalbeat
9/01 17:53:51.082933 async.go:63: INFO Flush Interval set to: 1s
9/01 17:53:51.082941 async.go:64: INFO Max Bulk Size set to: 2048
9/01 17:53:51.082947 async.go:71: DBG  create bulk processing worker (interval=1s, bulk size=2048)
9/01 17:53:51.083907 journalbeat.go:52: INFO Seek to tail successful
9/01 17:53:51.083980 beat.go:221: INFO journalbeat start running.
9/01 17:53:51.084057 journalbeat.go:183: INFO Journalbeat is running!
9/01 17:53:51.084110 journalbeat.go:201: WARN could not read the pending queue: EOF
9/01 17:53:51.084304 metrics.go:23: INFO Metrics logging every 30s
d assembly-journalbeat container.
y-journalbeat.service: main process exited, code=exited, status=2/INVALIDARGUMENT
sembly-journalbeat.service entered failed state.
y-journalbeat.service failed.
y-journalbeat.service holdoff time over, scheduling restart.
g assembly-journalbeat container...
ner "assembly-journalbeat" does not exist
 assembly-journalbeat container.
9/01 17:53:52.581532 beat.go:285: INFO Home path: [/] Config path: [/etc/journalbeat] Data path: [/var/lib/journalb
9/01 17:53:52.581566 beat.go:186: INFO Setup Beat: journalbeat; Version: 5.5.1
9/01 17:53:52.581578 processor.go:44: DBG  Processors:
9/01 17:53:52.581587 beat.go:192: DBG  Initializing output plugins
9/01 17:53:52.581620 outputs.go:108: INFO Activated console as output plugin.
9/01 17:53:52.581631 publish.go:238: DBG  Create output worker
9/01 17:53:52.581661 publish.go:280: DBG  No output is defined to store the topology. The server fields might not b
9/01 17:53:52.581700 publish.go:295: INFO Publisher name: journalbeat
9/01 17:53:52.582135 metrics.go:23: INFO Metrics logging every 30s
9/01 17:53:52.582861 async.go:63: INFO Flush Interval set to: 1s
9/01 17:53:52.582877 async.go:64: INFO Max Bulk Size set to: 2048
9/01 17:53:52.582888 async.go:71: DBG  create bulk processing worker (interval=1s, bulk size=2048)
9/01 17:53:52.583867 journalbeat.go:52: INFO Seek to tail successful
9/01 17:53:52.583965 beat.go:221: INFO journalbeat start running.
9/01 17:53:52.583977 journalbeat.go:183: INFO Journalbeat is running!
9/01 17:53:52.584014 journalbeat.go:201: WARN could not read the pending queue: EOF
9/01 17:54:22.582733 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:54:52.585736 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:55:22.582698 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:55:52.689932 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:56:22.586542 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:56:52.599777 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:57:22.601951 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:57:52.585381 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:58:22.594348 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:58:52.590427 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:59:22.613660 metrics.go:34: INFO No non-zero metrics in the last 30s
9/01 17:59:52.587451 metrics.go:34: INFO No non-zero metrics in the last 30s
```

**Comment by c4milo** [comment]
I finally got it working. The group I was using to run the container didn't have permissions to read `/run/log/journal/<machine_id>` Fixing that by assigning the GID corresponding to `systemd-journal` did it.

**Comment by vishr** [comment]
I just got the latest docker image and it worked and I don't know the reason.

**Comment by mheese** [comment]
@vishr the reason is most likely explained in #106 and fixed with #107 in case you are interested.

**Comment by vishr** [comment]
Sure thanks.
